Source/missiles.cpp

```diff
@@ -267,8 +269,10 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 
 	if (!shift)
 		dam <<= 6;
-	if (resist)
-		dam >>= 2;
+
+	assert(dam < (0x7FFFFFFF >> 6));
+	if (resist > 0)
+		dam = dam * (100 - resist) / 100;

```

Comment:
I find this a bit less convoluted:
```suggestion
		dam -= dam * resist / 100;
```

Reply:
I don't think those are equivalent? If you have 50 damage at 75% resist, the original would produce 12 damage, but I think the revised version would result in 13 damage (please check my math, I could be wrong)? I was trying to keep the result of 75% resist the same.